### PR TITLE
Add an example of xkcd scientific papers

### PR DESCRIPTION
**Pull request by jlowin**
image
image

### CODE REVIEW

File: docs/examples/xkcd_scientific_papers.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    paper_types = marvin.beta.extract(
+        data=marvin.beta.Image(
+            "https://imgs.xkcd.com/comics/types_of_scientific_paper_2x.png"
+        ), 
+        instructions="Extract the types of papers from the comic",
+    )
+
+    
+    # classify the paper
+    paper_text = load_paper("https://arxiv.org/pdf/1706.03762.pdf")
```
Comment (zzstoatzz) [inline docs/examples/xkcd_scientific_papers.md:24]: should `load_paper` be defined here or is this meant to be a placeholder?

**Review by zzstoatzz — APPROVED** [review]
cool!

File: docs/examples/xkcd_scientific_papers.md (hunk 1 above)
Comment (jlowin) [reply docs/examples/xkcd_scientific_papers.md:24]: placeholder, don't want to crowd the example. I will add a note.

**Review by zzstoatzz — APPROVED** [review]

**Review by jsugg — CHANGES_REQUESTED** [review]
# Suggestions
## Validation and Sanitization
Ensure that inputs (e.g., image paths and URLs) are validated, especially if they are user-provided.
## Error Handling Improvements
Implement detailed error handling around external API calls to manage failures or unexpected responses effectively.

File: docs/examples/xkcd_scientific_papers.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    # Open the PDF from the fetched bytes
+    pdf_bytes = response.content
+    pdf_stream = fitz.open(stream=pdf_bytes, filetype="pdf")
+    # Extract text from each page
+    text = ""
+    for page in pdf_stream:
+        text += page.get_text()
+    pdf_stream.close()  # Close the PDF stream
+    return text
+```
```
Comment (jsugg) [inline docs/examples/xkcd_scientific_papers.md:53]: Type hints won't hurt here; they'll educate instead.

**Pull request closed without merging**